cdm/core/src/test/java/ucar/nc2/ft2/coverage/TestCoverageCoordAxis1D.java

```diff
@@ -0,0 +1,52 @@
+package ucar.nc2.ft2.coverage;
+
+import org.junit.Test;
+import ucar.ma2.DataType;
+import ucar.nc2.constants.AxisType;
+import ucar.nc2.time.CalendarDate;
+import ucar.nc2.time.CalendarPeriod;
+import ucar.nc2.util.Optional;
+import static com.google.common.truth.Truth.assertThat;
+
+
+public class TestCoverageCoordAxis1D {
+
+  private static final CalendarDate testDate = CalendarDate.present();
+
+  private final int timeDelta = 2;
+
+  @Test
+  public void TestSubsetPresentTime1D() {
+    SubsetParams presentSubsetParams = new SubsetParams();
+    presentSubsetParams.setTimePresent();
+
+    CoverageCoordAxis1D timeAxis = create1DTimeAxis();
+    Optional<CoverageCoordAxis> subsetTimeAxis = timeAxis.subset(presentSubsetParams);
+
+    assertThat(subsetTimeAxis.isPresent()).isTrue();
+
+    if (!subsetTimeAxis.isPresent()) {
+      return;
```

Comment:
doesn't this not do anything if we made it through the first assert?

Reply:
I added the condition to avoid NPEs on the later asserts since the subsetTimeAxis is optional

Reply:
I think if the optional is empty then the test will fail on line 26 and not reach this condition

Reply:
Ah yeah, that makes a lot of sense! Sorry Optionals are new to me :)